### PR TITLE
fixed #104

### LINKED ISSUE #104: The cpp server trace is always broken
I tested the Cpp client to access the Java service, and everything was OK, But from the Cpp client to the Cpp service was always broken.

image

image

image

sample_op3 traceId is wrong,It is always recreated.

### PR DESCRIPTION
**Pull request by wuwen5**
fixed #104

image

### CODE REVIEW

**Comment by wuwen5** [comment]
It should have been broken here

https://github.com/SkyAPM/cpp2sky/commit/e5d747f0244067d25a1943da2a1177d5a571bd41#diff-c36c7b341b8772cdeb7c6252bd87482837ed9e287897fee7a64929b2b5bca36dR142

@wbpcode

File: source/tracing_context_impl.cc [hunk 1]
```diff
@@ -138,7 +138,7 @@ TracingContextImpl::TracingContextImpl(
     SpanContextExtensionPtr parent_ext_span_context, RandomGenerator& random)
     : parent_span_context_(std::move(parent_span_context)),
       parent_ext_span_context_(std::move(parent_ext_span_context)) {
-  segment_store_.set_traceid(random.uuid());
+  segment_store_.set_traceid(parent_span_context_->traceId());
```
Comment (wu-sheng) [inline source/tracing_context_impl.cc:141]: It should be parent trace ID, if not exist parent trace ID, create a UUID?

**Comment by wu-sheng** [comment]
@wuwen5 You break the CI.

File: source/tracing_context_impl.cc (hunk 1 above)
Comment (wbpcode) [reply source/tracing_context_impl.cc:141]: Thanks. But may be we should check if the parent context is existed or not?

```
segment_store_.set_traceid(parent_span_context_ ? parent_span_context_->traceId() : random.uuid());
```

**Comment by wbpcode** [comment]
And I think we should a test case to guard it. Thanks.

**Comment by wuwen5** [comment]
```cpp2sky test / e2e-python (pull_request) Failing```
This should be a network problem, The previous tests were all good.
I changed the code format for the last time

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**